grakn-engine/src/main/java/ai/grakn/engine/controller/api/AttributeTypeController.java

```diff
@@ -0,0 +1,144 @@
+import static ai.grakn.util.REST.WebPath.Api.ATTRIBUTE_TYPE;
+
+/**
+ * <p>
+ *     AttributeType endpoints
+ * </p>
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class AttributeTypeController {
+    private final EngineGraknTxFactory factory;
+    private static final Logger LOG = LoggerFactory.getLogger(AttributeTypeController.class);
+
+    public AttributeTypeController(EngineGraknTxFactory factory, Service spark) {
+        this.factory = factory;
+        spark.post(ATTRIBUTE_TYPE, this::postAttributeType);
+        spark.get(ATTRIBUTE_TYPE + "/" + ATTRIBUTE_TYPE_LABEL_PARAMETER, this::getAttributeType);
+    }
+
+    private Json postAttributeType(Request request, Response response) {
+        LOG.info("postAttributeType - request received.");
+        Json requestBody = Json.read(mandatoryBody(request));
+        String attributeTypeLabel = requestBody.at("attributeType").at("label").asString();
+        String attributeTypeDataTypeRaw = requestBody.at("attributeType").at("type").asString();
+        AttributeType.DataType<?> attributeTypeDataType = fromString(attributeTypeDataTypeRaw);
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("postAttributeType - attempting to add new attributeType " + attributeTypeLabel + " of type " + attributeTypeDataTypeRaw);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.WRITE)) {
+            AttributeType attributeType = tx.putAttributeType(attributeTypeLabel, attributeTypeDataType);
+            tx.commit();
+            String jsonConceptId = attributeType.getId().getValue();
+            String jsonAttributeTypeLabel = attributeType.getLabel().getValue();
+            String jsonAttributeTypeDataType = toString(attributeType.getDataType());
+            LOG.info("postAttributeType - attribute type " + jsonAttributeTypeLabel +
+                " of type " + jsonAttributeTypeDataType + " with id " + jsonConceptId + " added successfully. request processed.");
+            response.status(HttpStatus.SC_OK);
+            Json responseBody = attributeTypeJson(jsonConceptId, jsonAttributeTypeLabel, jsonAttributeTypeDataType);
+            return responseBody;
+        }
+    }
+
+    private Json getAttributeType(Request request, Response response) {
+        LOG.info("getAttributeType - request received.");
+        String attributeTypeLabel = mandatoryPathParameter(request, "attributeTypeLabel");
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("getAttributeType - attempting to find attributeType " + attributeTypeLabel + " in keyspace " + keyspace);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.READ)) {
+            Optional<AttributeType> attributeType = Optional.ofNullable(tx.getAttributeType(attributeTypeLabel));
+            if (attributeType.isPresent()) {
+                String jsonConceptId = attributeType.get().getId().getValue();
```

Comment:
It breaks my heart to see all this deserialisation code we can't so any better than this at the moment :(